api/views/content.py

```diff
@@ -144,5 +145,14 @@ class PublicPostsViewSet(
 ):
     renderer_classes = (PostsJSONRenderer,)
 
-    def get_queryset(self):
-        return self.queryset.filter(visibility="PUBLIC")
+    def list(self, request):
+        internal_posts = Post.objects.filter(visibility="PUBLIC")
+        serializer = PostSerializer(internal_posts, many=True)
+        posts = serializer.data
+
+        # dont return public posts of other nodes in node-to-node calls
+        if request.auth is not None and request.user.type is not "Node":
```

Comment:
If the user isn't authenticated then request.user is an insttance of `django.contrib.auth.models.AnonymousUser` which should still return public posts of other nodes.